src/game.rs

```diff
@@ -3092,6 +3203,167 @@ mod tests {
         assert!(game.is_err());
     }
 
+    #[test]
+    fn remove_lines() {
+        use crate::bet_size::BetSizeCandidates;
+        let card_config = CardConfig {
+            range: ["AA".parse().unwrap(), "AA".parse().unwrap()],
+            flop: flop_from_str("2c6dTh").unwrap(),
+            turn: card_from_str("3c").unwrap(),
+            ..Default::default()
+        };
+
+        // Simple tree: force checks until river, where OOP can bet 1/2 pot
+        let tree_config = TreeConfig {
+            initial_state: BoardState::Turn,
+            starting_pot: 60,
+            effective_stack: 970,
+            river_bet_sizes: [
+                BetSizeCandidates::try_from(("50%", "")).unwrap(),
+                Default::default(),
+            ],
+            ..Default::default()
+        };
+
+        let action_tree = ActionTree::new(tree_config).unwrap();
+        let mut game = PostFlopGame::with_config(card_config, action_tree).unwrap();
+
+        let lines = vec![vec![
+            Action::Check,
+            Action::Check,
+            Action::Chance(2),
+            Action::Bet(30),
+        ]];
+
+        let res = game.remove_lines(&lines);
+        assert!(res.is_ok());
+
+        game.allocate_memory(false);
+
+        // Check that the removed line
+        game.back_to_root();
+        game.play(game.node().actions.binary_search(&Action::Check).unwrap());
+        game.play(game.node().actions.binary_search(&Action::Check).unwrap());
+        game.play(2);
+        assert!(game.available_actions() == &[Action::Check]);
+        assert!(game.node().children.len() == 1);
```

Comment:
I prefer `assert_eq!` instead of `assert!` when available (omit comments on other lines).